### PR TITLE
The crate rename

### PR DESCRIPTION
**Pull request by gnunicorn**
refs #4099 .

Rename crates in `/primitives` and `/client`: prefix them with with `sp-`and `sc-` respectively, remove any unnecessary suffixing `-primitives`, `-runtime-api`. Fixup all the used names inside the repo.

## Upgrading

You'll find the bash script `rename-crates-for-2.0.sh` in the `.maintain`-folder. It is the script that was run and should be working the same way on your substrate-github-referencing project by renaming all the references in `Cargo.toml` and `*.rs`-files. You might have to `cargo clean` and build on a clean project again, if you encounter errors.

### CODE REVIEW

**Comment by gnunicorn** [comment]
force pushed a new run of the script against master (rather than attempting to fix the enormous conflict)

File: .maintain/rename-crates-for-2.0.sh [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    "substrate-network sc-network"
+    "substrate-offchain sc-offchain"
+    "substrate-peerset sc-peerset"
+    "substrate-rpc-servers sc-rpc-server"
+    "substrate-rpc sc-rpc"
+    "substrate-service sc-service"
+    "substrate-service-test sc-service-test"
+    "substrate-state-db sc-state-db"
+    "substrate-telemetry sc-telemetry"
+    "substrate-tracing sc-transaction"
```
Comment (tomusdrw) [inline .maintain/rename-crates-for-2.0.sh]: ?
```suggestion
    "substrate-tracing sc-tracing"
```

File: .maintain/rename-crates-for-2.0.sh [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    "sr-arithmetic sp-arithmetic"
+    "sr-arithmetic-fuzzer sp-arithmetic-fuzzer"
+    "sr-io sp-io"
+    "sr-primitives sp-runtime"
+    "sr-sandbox sp-sandbox"
+    "sr-staking-primitives sp-staking"
+    "sr-std sp-std"
+    "sr-version sp-version"
+    "substrate-state-machine sp-state-machine"
+    "substrate-transaction-pool-runtime-api sp-transaction-pool"
```
Comment (tomusdrw) [inline .maintain/rename-crates-for-2.0.sh]: This has been changed in #3833 it's now `sp-transaction-pool-runtime-api` already, we also have `sp-transaction-pool-api` and I believe that both should be merged into `sp-transaction-pool`, but let's do that in a separate PR. I guess this line can be removed from the script then as it has no effect, unless we want to help migrating other projects too.

Comment (gnunicorn) [reply .maintain/rename-crates-for-2.0.sh]: I'd leave it in, it does no harm and makes conversions for others idea. I just applied it against a clean master this morning and no bad side effects.
